### PR TITLE
feature: multiple certificates support for single domain

### PR DESCRIPTION
**Pull request by tokers**
### What this PR does / why we need it:

This PR solved issue https://github.com/apache/apisix/issues/1159, which proposes to support multiple certificates for a single domain.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by membphis** [comment]
failed in CI: https://github.com/apache/apisix/pull/2089/checks?check_run_id=1006023803#step:6:209

@tokers please take a look at the output of CI.

**Comment by tokers** [comment]
@membphis Fixed. I will add the document later.

File: t/router/radixtree-sni.t [hunk 1]
```diff
@@ -940,3 +940,134 @@ connected: 1
...
+
+            local ok, err = sock:connect("unix:$TEST_NGINX_HTML_DIR/nginx.sock")
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            ngx.say("connected: ", ok)
+
+            local sess, err = sock:sslhandshake(nil, "test3.com", false)
```
Comment (membphis) [inline t/router/radixtree-sni.t:1015]: Which certificate's SNI is `test3.com`?

Comment (membphis) [reply t/router/radixtree-sni.t:1015]: the SNI of `apisix_ecc` is `test.com`

Comment (tokers) [reply t/router/radixtree-sni.t:1015]: OK, i will fix it, just write this fake domain readily.

**Review by moonming — APPROVED** [review]

**Pull request merged**